airflow-core/src/airflow/utils/memray_utils.py

```diff
@@ -0,0 +1,84 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import logging
+from collections.abc import Callable
+from enum import Enum
+from functools import wraps
+from typing import ParamSpec, TypeVar
+
+from airflow.configuration import AIRFLOW_HOME, conf
+
+# Type variables for preserving function signatures
+PS = ParamSpec("PS")
+RT = TypeVar("RT")
+
+log = logging.getLogger(__name__)
+
+
+class MemrayTraceComponents(Enum):
+    """Possible airflow components can apply memray trace."""
+
+    scheduler = "scheduler"
+    dag_processor = "dag_processor"
+    api = "api"
+
+
+def enable_memray_trace(component: MemrayTraceComponents) -> Callable[[Callable[PS, RT]], Callable[PS, RT]]:
+    """
+    Conditionally track memory using memray based on configuration.
+
+    Args:
+        component: Enum value of the component for configuration lookup
+    """
+
+    def decorator(func: Callable[PS, RT]) -> Callable[PS, RT]:
+        @wraps(func)
+        def wrapper(*args: PS.args, **kwargs: PS.kwargs) -> RT:
+            _enable_memray_trace = conf.getboolean("profiling", "memray_trace_enabled")
+            if not _enable_memray_trace:
+                return func(*args, **kwargs)
+
+            _memray_trace_components = conf.getenumlist(
+                "profiling", "memray_trace_components", MemrayTraceComponents
+            )
```

Comment:
How about let just make it simpler by using `conf.get` to get the raw string and processing in the `memray_utils`?

```suggestion
            _memray_trace_components = conf.get(
                "profiling", "memray_trace_components"
            )
            # validating the string input
            # then parse to `MemrayTraceComponents` enum
            # ...
```